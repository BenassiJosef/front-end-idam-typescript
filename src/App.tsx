import React from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import Register from "./pages/Register/Register";
import WildCard from "./pages/WildCard/WildCard";
import GlobalStyle from "./components/GlobalStyle/GlobalStyle";

function App(): JSX.Element {
  return (
    <>
      <GlobalStyle />
      <div className="App">
        <Router>
          <Switch>
            <Route exact path="/" component={Register} />
            <Route exact path="/register" component={Register} />
            <Route component={WildCard} />
          </Switch>
        </Router>
      </div>
    </>
  );
}
export default App;
