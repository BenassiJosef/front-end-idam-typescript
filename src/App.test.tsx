import React from "react";
import { render, screen } from "@testing-library/react";
import App from "./App";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({});

test("renders learn react link", () => {
  render(
    <Provider store={store}>
      <App />
    </Provider>
  );
  const linkElement = screen.getByText(/Click Me/i);
  expect(linkElement).toBeInTheDocument();
});
