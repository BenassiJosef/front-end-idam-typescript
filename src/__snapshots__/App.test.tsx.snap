// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <header
          class="sc-gsTCUz cbApLE"
          color="#908f8f"
        >
          <div
            class="sc-hKgILt"
          >
            <h1
              class="sc-dlfnbm"
            >
              Register
            </h1>
          </div>
        </header>
        <div
          class="sc-pFZIQ eAnzZz"
        >
          <form
            class="sc-eCssSg dzZmmp"
            data-testid="form"
          >
            <div
              class="sc-iBPRYJ iuHtQb"
            >
              <label
                class="sc-gKsewC"
                for="name"
              >
                First Name
                <br />
                <input
                  aria-required="true"
                  class="sc-jSgupP tPSTZ"
                  data-testid="test-input"
                  id="name"
                  pattern=".{2,}"
                  required=""
                  title="More than one character"
                  type="text"
                  value=""
                />
              </label>
            </div>
            <div
              class="sc-iBPRYJ iuHtQb"
            >
              <label
                class="sc-gKsewC"
                for="familyName"
              >
                Second Name
                <br />
                <input
                  aria-required="true"
                  class="sc-jSgupP tPSTZ"
                  data-testid="test-input"
                  id="familyName"
                  pattern=".{2,}"
                  required=""
                  title="More than one character"
                  type="text"
                  value=""
                />
              </label>
            </div>
            <div
              class="sc-iBPRYJ iuHtQb"
            >
              <label
                class="sc-gKsewC"
                for="email"
              >
                Email
                <br />
                <input
                  aria-required="true"
                  class="sc-jSgupP tPSTZ"
                  data-testid="test-input"
                  id="email"
                  pattern=".{5,}"
                  required=""
                  title="Amount of character to construct email"
                  type="text"
                  value=""
                />
              </label>
            </div>
            <div
              class="sc-iBPRYJ iuHtQb"
            >
              <label
                class="sc-gKsewC"
                for="username"
              >
                Username
                <br />
                <input
                  aria-required="true"
                  class="sc-jSgupP tPSTZ"
                  data-testid="test-input"
                  id="username"
                  pattern=".{5,}"
                  required=""
                  title="Amount of character to construct email"
                  type="email"
                  value=""
                />
              </label>
            </div>
            <div
              class="sc-iBPRYJ iuHtQb"
            >
              <label
                class="sc-gKsewC"
                for="address"
              >
                Address
                <br />
                <input
                  aria-required="true"
                  class="sc-jSgupP tPSTZ"
                  data-testid="test-input"
                  id="address"
                  pattern=".{1,}"
                  required=""
                  title="Cannot be empty"
                  type="text"
                  value=""
                />
              </label>
            </div>
            <div
              class="sc-iBPRYJ iuHtQb"
            >
              <label
                class="sc-gKsewC"
                for="password"
              >
                Password
                <br />
                <input
                  aria-required="true"
                  class="sc-jSgupP tPSTZ"
                  data-testid="test-input"
                  id="password"
                  pattern=".{9,}"
                  required=""
                  title="8 characters minimum"
                  type="text"
                  value=""
                />
              </label>
            </div>
            <button
              class="sc-bdfBwQ dnaRID"
              data-test="primary-button"
              type="submit"
            >
               
              Submit
            </button>
          </form>
          <p>
            Already have an account? 
            <a
              aria-label="Link to login"
              class="sc-fubCfw kCOjCy"
              href="/login"
            >
               
              Login
            </a>
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <header
        class="sc-gsTCUz cbApLE"
        color="#908f8f"
      >
        <div
          class="sc-hKgILt"
        >
          <h1
            class="sc-dlfnbm"
          >
            Register
          </h1>
        </div>
      </header>
      <div
        class="sc-pFZIQ eAnzZz"
      >
        <form
          class="sc-eCssSg dzZmmp"
          data-testid="form"
        >
          <div
            class="sc-iBPRYJ iuHtQb"
          >
            <label
              class="sc-gKsewC"
              for="name"
            >
              First Name
              <br />
              <input
                aria-required="true"
                class="sc-jSgupP tPSTZ"
                data-testid="test-input"
                id="name"
                pattern=".{2,}"
                required=""
                title="More than one character"
                type="text"
                value=""
              />
            </label>
          </div>
          <div
            class="sc-iBPRYJ iuHtQb"
          >
            <label
              class="sc-gKsewC"
              for="familyName"
            >
              Second Name
              <br />
              <input
                aria-required="true"
                class="sc-jSgupP tPSTZ"
                data-testid="test-input"
                id="familyName"
                pattern=".{2,}"
                required=""
                title="More than one character"
                type="text"
                value=""
              />
            </label>
          </div>
          <div
            class="sc-iBPRYJ iuHtQb"
          >
            <label
              class="sc-gKsewC"
              for="email"
            >
              Email
              <br />
              <input
                aria-required="true"
                class="sc-jSgupP tPSTZ"
                data-testid="test-input"
                id="email"
                pattern=".{5,}"
                required=""
                title="Amount of character to construct email"
                type="text"
                value=""
              />
            </label>
          </div>
          <div
            class="sc-iBPRYJ iuHtQb"
          >
            <label
              class="sc-gKsewC"
              for="username"
            >
              Username
              <br />
              <input
                aria-required="true"
                class="sc-jSgupP tPSTZ"
                data-testid="test-input"
                id="username"
                pattern=".{5,}"
                required=""
                title="Amount of character to construct email"
                type="email"
                value=""
              />
            </label>
          </div>
          <div
            class="sc-iBPRYJ iuHtQb"
          >
            <label
              class="sc-gKsewC"
              for="address"
            >
              Address
              <br />
              <input
                aria-required="true"
                class="sc-jSgupP tPSTZ"
                data-testid="test-input"
                id="address"
                pattern=".{1,}"
                required=""
                title="Cannot be empty"
                type="text"
                value=""
              />
            </label>
          </div>
          <div
            class="sc-iBPRYJ iuHtQb"
          >
            <label
              class="sc-gKsewC"
              for="password"
            >
              Password
              <br />
              <input
                aria-required="true"
                class="sc-jSgupP tPSTZ"
                data-testid="test-input"
                id="password"
                pattern=".{9,}"
                required=""
                title="8 characters minimum"
                type="text"
                value=""
              />
            </label>
          </div>
          <button
            class="sc-bdfBwQ dnaRID"
            data-test="primary-button"
            type="submit"
          >
             
            Submit
          </button>
        </form>
        <p>
          Already have an account? 
          <a
            aria-label="Link to login"
            class="sc-fubCfw kCOjCy"
            href="/login"
          >
             
            Login
          </a>
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
